### PR TITLE
SVM: Move nonce_info from accounts-db to Solana SDK

### PR DESCRIPTION
**Pull request by dmakarov**
#### Problem

Continue decoupling SVM from accounts-db.

#### Summary of Changes

Move nonce_info module from accounts-db to Solana SDK and fix up corresponding imports.

### CODE REVIEW

File: sdk/src/nonce_info.rs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-use solana_sdk::{
+use crate::{
```
Comment (pgarg66) [inline sdk/src/nonce_info.rs:1]: You likely need to add the following at the top to get past the sbf errors.

```
#![cfg(feature = "full")]
```

**Review by pgarg66 — APPROVED** [review]

**Comment by jstarry** [comment]
I think this makes more sense to live in `solana-svm` than `solana-sdk` because it's not something we want to be subjected to the backwards compatibility policy. @dmakarov @pgarg66 can you move this out of sdk?

**Comment by dmakarov** [comment]
Should nonce_account also be moved to SVM?

Also moving nonce_info to SVM will make rpc and accounts-db depend on SVM — is this desirable?

**Comment by pgarg66** [comment]
> Should nonce_account also be moved to SVM?
> 
> 
> 
> Also moving nonce_info to SVM will make rpc and accounts-db depend on SVM — is this desirable?

I think downwards dependency is ok. The reverse would have been a problem.

**Comment by jstarry** [comment]
> Should nonce_account also be moved to SVM?

Just `nonce_info` for now.. looks like `nonce_account` has been in sdk for awhile so technically we may have to deprecate it first

**Pull request merged**